pyodata/client.py

```diff
@@ -37,6 +53,25 @@ class Client:
 
     ODATA_VERSION_2 = 2
 
+    @staticmethod
+    async def build_async_client(url, connection, odata_version=ODATA_VERSION_2, namespaces=None,
+                                 config: pyodata.v2.model.Config = None, metadata: str = None):
+        """Create instance of the OData Client for given URL"""
+
+        logger = logging.getLogger('pyodata.client')
+
+        if odata_version == Client.ODATA_VERSION_2:
+
+            # sanitize url
+            url = url.rstrip('/') + '/'
+
+            if metadata is None:
+                metadata = await _async_fetch_metadata(connection, url, logger)
```

Comment:
Don't you kill all async here? What's the difference between the regular client?

Reply:
no, await (asynchronous wait) is calling the asynchronous _async_fetch_metadata function.